From dfcc89c6b878a53cad587cb5dda63579d969e9c5 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 17 Sep 2024 09:46:12 +0200
Subject: [PATCH 1236/1695] dts: stm32: refine STM32MP21 secure and non-secure
 USART2 pinctrl states

Explicitly state that legacy pinctrl phandles usart2_pins_a
refer to non-secure USART2 pin muxing, used in STM32MP21 based
boards for OP-TEE console using a non-secure UART bus.

Define secure USART2 bus pinctrl states for board that needs
to use the USART2 bus in secure state.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git abd594a878052747d364ed64a67ad609d686d83f dts: stm32: refine STM32MP21 secure and non-secure USART2 pinctrl states]
Change-Id: I9020607ccd2cdd9fd45522ca928a48576a2429b8
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404815
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp21-pinctrl.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
index 8090af4a3..6ef7edc58 100644
--- a/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
+++ b/core/arch/arm/dts/stm32mp21-pinctrl.dtsi
@@ -18,6 +18,19 @@
 	};
 
 	usart2_pins_a: usart2-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX_NSEC('A', 4, AF6)>; /* USART2_TX */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <0>;
+		};
+		pins2 {
+			pinmux = <STM32_PINMUX_NSEC('A', 8, AF8)>; /* USART2_RX */
+			bias-disable;
+		};
+	};
+
+	usart2_secure_pins_a: usart2-sec-0 {
 		pins1 {
 			pinmux = <STM32_PINMUX('A', 4, AF6)>; /* USART2_TX */
 			bias-disable;
-- 
2.47.2

